### PR TITLE
[#166572994] Fix stop polling

### PR DESCRIPTION
**Pull request by Edigio**
Fix stop polling

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🐞 [#166572994](https://www.pivotaltracker.com/story/show/166572994): Durante l'attiva, annullando il processo di pagamento, la saga che fa il polling dell'idPayment non viene interrotta

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d440666d4805928bfc8b21c6fe19199e71aac2ae
</p>

**Comment by Undermaken** [comment]
I'm just reading the code comments and:

> // Whether we must stop retrying
> // the abort process gets triggered when the shouldAbort promise resolves
> // to true. Not that aborting stops the retry process, it does NOT stop
> // the execution of the current task.

(I guess _**Not**_ is _**Note**_)

Anyway cloud a caller use _**shouldAbort**_ promise to stop retry execution?

**Comment by Edigio** [comment]
> I'm just reading the code comments and:
> 
> > // Whether we must stop retrying
> > // the abort process gets triggered when the shouldAbort promise resolves
> > // to true. Not that aborting stops the retry process, it does NOT stop
> > // the execution of the current task.
> 
> (I guess _**Not**_ is _**Note**_)
> 
> Anyway cloud a caller use _**shouldAbort**_ promise to stop retry execution?

 _**shouldAbort**_ , if I'm not mistaken, is used to abort a single task after timeout but not the retry cycle. Anyway i have created another Promise to abort the cycle of requests

File: src/tasks.ts [hunk 1]
```diff
@@ -43,14 +48,26 @@ export type RetriableTask<E, T> = E extends TransientError
...
   maxRetries: number,
-  backoff: (count: number) => Millisecond
+  backoff: (count: number) => Millisecond,
+  controllerRetry?: Promise<Response>
 ): (
   _: RetriableTask<E, T>,
   shouldAbort?: Promise<boolean>
 ) => TaskEither<E | MaxRetries | RetryAborted, T> {
   return (task, shouldAbort = Promise.resolve(false)) => {
+    // tslint:disable-next-line: no-let
```
Comment (cloudify) [inline src/tasks.ts]: this is basically a duplication of the logic implemented by passing `shouldAbort`, the fix should be done by passing `shouldAbort` [here](https://github.com/teamdigitale/io-ts-commons/blob/master/src/fetch.ts#L126)

**Comment by cloudify** [comment]
@Edigio check https://github.com/teamdigitale/io-ts-commons/pull/250

**Comment by cloudify** [comment]
Superseded by #250

**Pull request closed without merging**